modules/collections/client/controllers/collections.controllers.js

```diff
@@ -1,139 +1,154 @@
-        var originalDocuments = _.map(sourceDocuments, function(cd) { return cd.document; });
-
-        // Find documents added to the collection
-        var addedDocuments = _.filter(updatedDocuments, function(updatedDoc) {
-          return !_.find(originalDocuments, function(originalDoc) { return originalDoc._id === updatedDoc._id; });
-        });
-
-        // Find documents removed from the collection
-        var removedDocuments = _.filter(originalDocuments, function(originalDoc) {
-          return !_.find(updatedDocuments, function(updatedDoc) { return updatedDoc._id === originalDoc._id; });
-        });
-
-        // Generate promise for document management
-        var docPromises = _.union(_.map(addedDocuments, function(doc) {
-          return CollectionModel.addDocument(collection._id, doc._id, docType);
-        }), _.map(removedDocuments, function(doc) {
-          return CollectionModel.removeDocument(collection._id, doc._id, docType);
-        }));
-
-        // complete the task
-        Promise.all(docPromises)
-          .then(reloadEdit)
-          .catch(function(/* res */) {
-            AlertService.error('Could not update other documents for "'+ $scope.collection.displayName +'".', 4000, true);
-            reloadEdit();
-          });
-      };
+      // Checks the validity of the collection's input fields.
+      $scope.checkValidity = function(publishing) {
+        $scope.validationFlags.detailsTabInvalid = false;
+        $scope.validationFlags.documentsTabInvalid = false;
+        // This resets validity.
+        $scope.validationFlags.save = true;
+        // Only start checking validation if user tries to save or publish.
+        $scope.checkDetailsField();
+        if (publishing) {
+          $scope.checkDocumentsField();
+        }
+      }
 
-      $scope.removeDocument = function(document, type) {
-        CollectionModel.removeDocument($scope.collection._id, document._id, type)
-          .then(reloadEdit)
-          .catch(function(/* res */) {
-            AlertService.error('Could not remove document from "'+ $scope.collection.displayName +'".', 4000, true);
-            reloadEdit();
+      // Helper function to check validity of Details tab specifically.
+      $scope.checkDetailsField = function() {
+        if ($scope.validationFlags.save) {
+          $scope.validationFlags.name = $scope.collection.displayName.length < 1 ? true : false;
+          var duplicateName = _.find($scope.$parent.$resolve.collections, function(collection) {
+            return ((collection.displayName == $scope.collection.displayName) && (collection.id != $scope.collection.id));
           });
-      };
+          if (duplicateName) {
+            $scope.validationFlags.name = true;
+            AlertService.error('Project already contains a collection named "'+ $scope.collection.displayName +'".', 4000, true)
+          }
+          $scope.validationFlags.type = $scope.collection.type.length < 1 ? true : false;
```

Comment:
No need to use a ternary operator here, `$scope.collection.type.length < 1` will already return true or false. The same should apply for the next flags as well.

Reply:
Good call. I was unsure how booleans are evaluated in JS, so I was playing it safe. It's one of those things I forgot to check when cleaning up my code at the end.